compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DocumentationAnalyzer.java

```diff
@@ -167,6 +211,117 @@ public void visit(BLangResource resourceNode) {
                 DiagnosticCode.PARAMETER_ALREADY_DOCUMENTED);
 
         validateReturnParameter(resourceNode, null, false);
+        validateReferences(resourceNode);
+    }
+
+    private void validateReferences(DocumentableNode documentableNode) {
+        BLangMarkdownDocumentation documentation = documentableNode.getMarkdownDocumentationAttachment();
+        if (documentation == null) {
+            return;
+        }
+
+        Stack<BLangMarkdownReferenceDocumentation> references = documentation.getReferences();
+        references.forEach(reference -> {
+            boolean isValidIdentifier = validateIdentifier(reference.getPosition(), documentableNode,
```

Comment:
Better to pass the `reference` and `documentableNode` to `validateIdentifier` method rather than passing various attributes of the reference separately. Within the method, these attributes can be assigned to new variables if required.